app/org/sagebionetworks/bridge/services/AuthenticationService.java

```diff
@@ -304,7 +298,8 @@ private UserSession getSessionFromAccount(Study study, CriteriaContext context,
         if (participant.getLanguages().isEmpty() && !context.getLanguages().isEmpty()) {
             participant = new StudyParticipant.Builder().copyOf(participant)
                     .withLanguages(context.getLanguages()).build();
-            optionsService.setOrderedStringSet(study, account.getHealthCode(), LANGUAGES, context.getLanguages());
+            account.setLanguages(context.getLanguages());
+            accountDao.updateAccount(account, false);
```

Comment:
I'm not convinced this won't fail if this is the same request that, for example, verifies a channel. I think we should use editAccount() here.

Reply:
Okay, changed to editAccount